RoddyCore/src/de/dkfz/roddy/core/CohortDataRuntimeServiceExtension.groovy

```diff
@@ -125,14 +136,93 @@ class CohortDataRuntimeServiceExtension {
         }
         return datasets as List<DataSet>
     }
 
+    /**
+     * Returns true, if all filters are valid super cohort identifiers.
+     * @return
+     */
+    boolean checkIfListOnlyContainsSCIdentifiers(List<String> pidFilters) {
+        // We use the COHORT_IDENTIFIER. Rules for it also apply for the super cohort identifier.
+        pidFilters.every { it ==~ COHORT_IDENTIFIER }
+    }
+
+    /**
+     * 1. Check, if super cohort cvalue is present.
+     * 2. Check, if super cohort cvalue is valid.
+     * 3. Check for all cohort identifiers in a super cohort cvalue, if they are present.
+     * 4. Check for all cohort cvalues, if they are valid.
+     * @param configuration
+     * @param scIdentifiers
+     */
+    void validateSuperCohortCValuesOrFail(Configuration configuration, List<String> scIdentifiers) {
+        def values = configuration.configurationValues.allValues
+
+        // 1. Check, if super cohort cvalue is present.
+        def unavailableSuperCohorts = scIdentifiers.findAll() { !configuration.configurationValues.hasValue(it) }
+        if (unavailableSuperCohorts)
+            throw new ConfigurationError((["One or more requested super cohorts are not present in the configuration:"] + unavailableSuperCohorts).join("\n\t"), "Super cohort description missing")
+
+        // 2. Check, if super cohort cvalue is valid.
+        Map<String, String> cohortStrings = scIdentifiers.collectEntries { [it, values[it]?.value] }
+        List<String> invalidCohortListMessages = cohortStrings.collect { String sc, String list ->
+            boolean match = list ==~ "${COHORT_IDENTIFIER}([,]${COHORT_IDENTIFIER}){0,}"
+            return match ? "" : "\t${sc} => ${list}".toString()
+        }.findAll()
+        if (invalidCohortListMessages)
+            throw new ConfigurationError((["One or more cohort strings are invalid:"] + invalidCohortListMessages).join("\n"), "Invalid cohort descriptor")
+
+        // 3. Check for all cohort identifiers in a super cohort cvalue, if they are present.
+        String missingCohorts = scIdentifiers.collect { String sc ->
+            cohortStrings[sc].split(StringConstants.SPLIT_COMMA).collect { String cohortID ->
+                if (!configuration.configurationValues.hasValue(cohortID))
+                    return "\tCohort cvalue ${cohortID} is missing"
+                return ""
+            }.findAll().join("\n")
+        }.findAll().join("\n")
+        if (missingCohorts)
+            throw new ConfigurationError("One or more cohort cvalues are not present in the configuration:\n" + missingCohorts, "Cohort description missing")
+
+        // 4. Check for all cohort cvalues, if they are valid.
+        String malformedPIDIDs = scIdentifiers.collect { String sc ->
+            cohortStrings[sc].split(StringConstants.SPLIT_COMMA).collect {
+                String cohortID ->
+                    if (!values[cohortID].value ==~ "${PID_IDENTIFIER}([,]${PID_IDENTIFIER}){0,}")
```

Comment:
I'm getting a Cohort cvalue malformed error here.

Used cohorts in the projectConfig XML
```
<cvalue name="cohortA" value="d11,d12,d13,d14" type="string"/>
<cvalue name="cohortB" value="ICGC_MB6,ICGC_MB7,ICGC_MB8,ICGC_MB9,ICGC_MB10" type="string"/>
``` 
Console Output:
```
de.dkfz.roddy.config.ConfigurationError: One or more cohort cvalues are malformed:
	supercohort, cohortA => d11,d12,d13,d14 is malformed
	supercohort, cohortB => ICGC_MB6,ICGC_MB7,ICGC_MB8,ICGC_MB9,ICGC_MB10 is malformed
	at de.dkfz.roddy.core.CohortDataRuntimeServiceExtension.validateSuperCohortCValuesOrFail(CohortDataRuntimeServiceExtension.groovy:202)
	at de.dkfz.roddy.core.CohortDataRuntimeServiceExtension.loadCohortDatasetsWithFilter(CohortDataRuntimeServiceExtension.groovy:122)
	at de.dkfz.roddy.core.RuntimeService.loadDatasetsWithFilter(RuntimeService.groovy:164)
	at de.dkfz.roddy.core.RuntimeService.loadDatasetsWithFilter(RuntimeService.groovy)
	at de.dkfz.roddy.core.Analysis.run(Analysis.java:190)
	at de.dkfz.roddy.client.cliclient.RoddyCLIClient.testrun(RoddyCLIClient.groovy:523)
	at de.dkfz.roddy.client.cliclient.RoddyCLIClient.parseStartupMode(RoddyCLIClient.groovy:119)
	at de.dkfz.roddy.Roddy.parseRoddyStartupModeAndRun(Roddy.java:737)
	at de.dkfz.roddy.Roddy.startup(Roddy.java:287)
	at de.dkfz.roddy.Roddy.main(Roddy.java:214)
```

Reply:
it might be that the "!" in `if (!values[cohortID].value ==~ "${PID_IDENTIFIER}([,]${PID_IDENTIFIER}){0,}")` has no effect and needs another set of parentheses:
```
if (!(values[cohortID].value ==~ "${PID_IDENTIFIER}([,]${PID_IDENTIFIER}){0,}"))
```

Reply:
You were right, this caused the problem. Interestingly though, the tests already cover it and worked. Fixed it.